src/plugin/vectortile/vectortilelayerconfig.js

```diff
@@ -0,0 +1,294 @@
+ * @param {Array<string>} fonts The style fonts.
+ * @return {Array<string>} The supported fonts.
+ */
+const getFonts = (fonts) => [DEFAULT_FONT];
+
+
+/**
+ */
+class VectorTileLayerConfig extends AbstractLayerConfig {
+  /**
+   * Constructor.
+   */
+  constructor() {
+    super();
+
+    /**
+     * @type {?CrossOrigin}
+     * @protected
+     */
+    this.crossOrigin = null;
+
+    /**
+     * @type {Projection}
+     * @protected
+     */
+    this.projection = null;
+
+    /**
+     * List of URLs for load balancing.
+     * @type {Array<string>}
+     * @protected
+     */
+    this.urls = [];
+  }
+
+  /**
+   * @inheritDoc
+   */
+  initializeConfig(options) {
+    super.initializeConfig(options);
+
+    if (options['urls'] != null) {
+      this.urls = /** @type {!Array<string>} */ (options['urls']);
+    } else if (this.url) {
+      // make sure the "urls" property is set in the options for multiple URL support
+      options['urls'] = this.urls = [this.url];
+
+      // remove the "url" property to avoid confusion
+      options['url'] = undefined;
+    }
+
+    this.expandUrls();
+    options['urls'] = this.urls;
+
+    const projection = getBestSupportedProjection(options);
+    if (!projection) {
+      throw new Error('No projections supported by the layer are defined!');
+    }
+
+    this.projection = projection;
+
+    // cross origin
+    if (!net.isValidCrossOrigin(options['crossOrigin'])) {
+      this.crossOrigin = /** @type {CrossOrigin} */ (net.getCrossOrigin(this.urls[0]));
+    } else {
+      this.crossOrigin = /** @type {CrossOrigin} */ (options['crossOrigin']);
+
+      for (let i = 0; i < this.urls.length; i++) {
+        const url = this.urls[i];
+
+        // register the cross origin value by URL pattern so that our Cesium.loadImage mixin can find it
+        net.registerCrossOrigin(AbstractTileLayerConfig.getUrlPattern(url), this.crossOrigin);
+      }
+    }
+
+    // the correct none equivalent for crossOrigin in OL is null
+    if (this.crossOrigin === CrossOrigin.NONE) {
+      this.crossOrigin = null;
+      options['crossOrigin'] = null;
+    }
+
+    // Default render mode to 'image'
+    if (!options['renderMode']) {
+      options['renderMode'] = VectorTileRenderType.IMAGE;
+    }
+
+    // If zoom levels aren't specified, assume vector tiles can be generated at all levels
+    if (options['minZoom'] == null) {
+      options['minZoom'] = DEFAULT_MIN_ZOOM;
+    }
+
+    if (options['maxZoom'] == null) {
+      options['maxZoom'] = DEFAULT_MAX_ZOOM;
+    }
+  }
+
+  /**
+   * @param {Object<string, *>} options The options
+   * @return {VectorTileSource}
+   */
+  getSource(options) {
+    // Copy the original options to avoid adding a non-serializable object to it.
+    const sourceOptions = Object.assign({}, options);
+
+    const vtFormat = /** @type {VectorTileFormat|undefined} */ (sourceOptions['format']);
+    sourceOptions['format'] = getVectorTileFormat(vtFormat || VectorTileFormat.MVT);
```

Comment:
Is this format property used by us or in the superclass for the `VectorTileSource`? Do we need it in quotes if its an externed options type?

Reply:
Good point. It's used by the OL class so `sourceOptions` will need a type cast, but `format` is defined on `olx.source.VectorTileOptions`.